linera-execution/src/system.rs

```diff
@@ -684,6 +674,64 @@ where
                 self.read_blob_content(blob_id).await?;
                 self.blob_used(Some(txn_tracker), blob_id).await?;
             }
+            ProcessNewEpoch(epoch) => {
+                let chain_next_epoch = self.epoch.get().expect("chain is active").try_add_one()?;
+                ensure!(
+                    epoch == chain_next_epoch,
+                    SystemExecutionError::InvalidCommitteeCreation
+                );
+                let admin_id = self
+                    .admin_id
+                    .get()
+                    .ok_or_else(|| SystemExecutionError::InactiveChain)?;
+                let event_id = EventId {
+                    chain_id: admin_id,
+                    stream_id: StreamId::system(EPOCH_STREAM_NAME),
+                    key: bcs::to_bytes(&epoch)?,
+                };
+                let bytes = match txn_tracker.next_replayed_oracle_response()? {
+                    None => self.context().extra().get_event(event_id.clone()).await?,
+                    Some(OracleResponse::Event(recorded_event_id, bytes))
+                        if recorded_event_id == event_id =>
+                    {
+                        bytes
+                    }
+                    Some(_) => return Err(SystemExecutionError::OracleResponseMismatch),
+                };
+                let committee = bcs::from_bytes(&bytes)?;
+                txn_tracker.add_oracle_response(OracleResponse::Event(event_id, bytes));
```

Comment:
Do we need the whole committee in the oracle event here? `EventId` should uniquely identify the committee.

Reply:
I guess we could put the committee into a blob, and make the event only contain the blob ID… :thinking:

Reply:
Or that. I think the committee in the event is unnecessary - it can be retrieved if needed but I don't think that's even the case right now - and it bloats the event size.

Reply:
You mean we could read the original admin chain block instead?

We could, but it's a bit more complicated (iterate over the operations to find the one with the committee), and reads more data (the whole blob instead of just the committee).

Another option is to store the committees separately in storage, as yet another type of data. Then we don't even need them in the chain states? :thinking:

Reply:
> You mean we could read the original admin chain block instead?

It's already in the storage, right? We just need to be able to get it but we already do - you're using it in this PR.

Reply:
> It's already in the storage, right?

Yes, as part of the admin chain block.

Would you prefer to only put the admin chain block hash into the event, and then load the block, and go through its iterations to find the operation that contains the committee?

Then we'd have to do a similar thing on the validator side, instead of using a mechanism that we'll use for all events.

And when we change the admin chain logic (e.g. using a staking contract), we would have to change that mechanism again, both for clients and validators, instead of just emitting an event from there.